camayoc/tests/qpc/api/v1/credentials/test_network_creds.py

```diff
@@ -36,18 +37,22 @@ def test_update_password_to_sshkeyfile(shared_client, cleanup, isolated_filesyst
         3) Confirm network credential has been updated.
     :expectedresults: The network credential is updated.
     """
-    cred = Credential(cred_type="network", client=shared_client, password=uuid4())
-    cred.create()
-    # add the id to the list to destroy after the test is done
-    cleanup.append(cred)
+    cred = next(
+        data_provider.credentials.new(
+            match_criteria={"type": "network", "password": Table.is_not_null()},
+            data_only=False,
+        )
+    )
     assert_matches_server(cred)
 
-    sshkeyfile_name = utils.uuid4()
-    tmp_dir = os.path.basename(os.getcwd())
-    sshkeyfile = Path(sshkeyfile_name)
-    sshkeyfile.touch()
+    ssh_network_cred_data = next(
+        data_provider.credentials.new(
+            match_criteria={"type": "network", "sshkeyfile": Table.is_not_null()},
+            data_only=True,
+        )
+    )
 
-    cred.ssh_keyfile = f"/sshkeys/{tmp_dir}/{sshkeyfile_name}"
+    cred.ssh_keyfile = ssh_network_cred_data.ssh_keyfile
```

Comment:
I'm really happy how this turned out. As long as we can assume that config contains at least one credential with valid ssh key, we can use DataProvider to generate an endless stream of different credentials using SSH key. This effectively turns DISCOVERY-284 from a blocker to wishlist item, and allows us to run entire suite sooner than I initially expected.